cargo-dist/templates/installer/installer.sh.j2

```diff
@@ -34,6 +34,49 @@ RECEIPT_HOME="${HOME}/.config/{{ app_name }}"
 BUILDER_GLIBC_MAJOR="2"
 BUILDER_GLIBC_SERIES="31"
 
+# We assign this right at the top of the file as a global
+# because a number of different things will try to read it.
+INSTALL_PATH_STRATEGY=""
+INSTALL_PATH_ENV_KEY=""
+INSTALL_PATH_SUBDIR=""
+INSTALL_PATH_HINT=""
+
+# If you're reading this install script generated by an app, and
+# this looks strange to you, it's because it's possible for there
+# to be multiple generated paths here. If you're only seeing one
+# $install_path assignment, it's because the app you're looking at
+# has been configured with a single install path.
+{% for install_path in install_paths %}
+if [ -z "$INSTALL_PATH_STRATEGY" ]; then
+{% if install_path.kind == "CargoHome" %}
+    # This strategy can't fail; always assign it
+    INSTALL_PATH_STRATEGY="CargoHome"
+    INSTALL_PATH_HINT="\$CARGO_HOME/bin (\$HOME/.cargo/bin)"
```

Comment:
Yeah i still think the install-hints shouldn't be this dynamic. If you ask the script what it does, it should report its actual configurable behaviour, and not just what it will do with the current set of env-vars. That said if we're printing out a list of options, I suppose I wouldn't complain if the printout showed the expression *and* its computed value.

Like

```text
then unpacks the binaries and installs them to the first defined option:
   $MY_COMPANY (unset)
   $MY_COMPANY_LEGACY/bin = /home/gankra/corpcrud/bin
   $HOME/.mycompany/bin = /home/gankra/.mycompany/bin
```

Reply:
That's reasonable, yeah. By comparison the Windows version *does* do that, so it would be better to have it be identical.